### DISCUSSION TITLE
Multi-user support for messaging

### DISCUSSION THREAD

**Discussion by Sanaki** [RetroAchievements/RAWeb#1555] [Ideas]
Currently we have a number of meta-accounts on the site, such as RAdmin and QATeam, which rely on credentials shared between a number of different users. Short term, I'd like for us to be able to authorize specific users to access those accounts for messaging purposes without having to log out of their personal account and log back in via shared credentials. Long term, it might be viable to let users manage their alt accounts in the same fashion. I'm uncertain if this is viable to implement with the current state of the site.

The way I envision it, when writing a PM, a user authorized for RAdmin for instance would be able to select which account the message comes from during the message-writing process. The inbox would similarly allow selecting which inbox is displayed by such authorized users.

**Comment by SporyTike** [comment]
Would like to revive this discussion as it is pretty important due to the growing community and the connected growing interactions of meta-teams.  Most inboxes from these teams are read via Discord inbox due to how we have things set up. So I am not sure how viable this feature per se is for normal users because usually you read your inbox by accessing the account so you are already logged in and usually answer with that account. (Another feature for quick account-swap like Discord has would be a more viable option.)
- - - - -
To get back to the option to send messages from another account: I would like to easily declare from what account my message is sent without needing to actually swap them but with a simply click. I think an easy way to implement that would be a simply dropdown that replaces the sender username/user ID like my example image. That way even users like site moderators who likely have access to more than 5 meta accounts don't get bloated. People who do not participate in a team with a meta account won't see the dropdown.
![image](https://github.com/RetroAchievements/RAWeb/assets/45672834/abfadeeb-054d-45e2-9c25-4b8c112d2926)

For the general question who has access to an account, I am unsure if the permission matrix is needed right now or not. My current idea how to tackle it is a bit complex while the matrix likely just would need a flag like "Is QATeam".
In my head a solution for now would be cloning the "Follower" feature and reversing the affect. So if RAdmin follows a user, that user gets access to the 'RAdmin' option of the dropdown. However, that would likely require a second requirement to ensure not any user from a meta team can add or remove other users for whatever reason.

**Comment by redwizard42** [comment]
Spory's Mock-up is about how I'd envision it being done from a user-perspective, but I worry it's not obvious? Then again, having it near the Send Message button is probably wise. "Send Message As..." would perhaps be a better label than "Account."

Questions/Thoughts:
1. Would any guarding be needed "Are you sure?"
2. Should that prompt always happen, even if sending as yourself, to make sure you didn't mean to send using a team account?
3. Would it be instead better to have the "What account do you wish to send from?" happen when you do to start a message (clicking on the message link for a user), to auto-populate that field in the message entry view?
4. Would this functionality also have a way to "Read <account> messages?" such that we can view the inbox for the team account from our own, and then it would default to sending from that account if you reply to a message?
